### ISSUE TITLE
DataServicesClient - MERGE/PATCH requests are identical to PUT

### ISSUE THREAD

**Issue by uffelauesen** [OData/odata.net#683]
_MERGE/PATCH means a partial update of only a part of an object (only some properties should be changed on server). PUT means replace the entire object on the server with what the client sends. DataServicesClient sends the entire object (all properties) for both MERGE/PATCH and PUT - now how would the server know the subset of properties that should be updated on MERGE/PATCH ?. It cannot know this unless the DataServicesClient only sends those properties that should be "patched". This could lead to loss of data in cases where the entity was originally selected with projection (most properties will then have null values) and then a MERGE/PATCH will submit those to the server - resulting in those unprojected properties being updated with null - OUCH. I do know you can fiddle with the request pipeline and do some magic your self, but we have a lot of external developers coding against our service - they probably "forget" to do such lowlevel magic._  
### Assemblies affected

_DataServicesClient V3 5.7.0._  
### Reproduce steps

*Do a query with projection selecting only a "Title" property - select it into a DSC<>. Update the "Title" property and call ctx.SaveChanges(). Inspect the generated request. \*  
### Expected result

_Some mechanism (like property change tracking) ensures that the server can correctly understand the clients intent of only update the "Title" property of the entity._  
### Actual result

_The server has no way of telling what the client intended - it states to do a MERGE/PATCH, but for what properties? The client send all of them._  
### Additional details

_This a very old issue, that was first reported back in 2011 [UserVoice](https://data.uservoice.com/forums/72027-wcf-data-services-feature-suggestions/suggestions/1436771-track-entity-changes-at-the-property-level). You have since then made this (rather simple) change to V4, it would be rather simple to downport this to V3 - still a lot of services and clients out there running V3. Please also see my comment from 2014: [TeamBlog](https://blogs.msdn.microsoft.com/odatateam/2014/04/09/tutorial-sample-client-property-tracking-for-patch/#comment-15503)._

**Comment by lichader** [comment]
Hi,

I just come into this issue. I am writing an app that will be sending data to dynamics CRM API. But for the entity change, I only updated one property and it sent all fields of this entity in the PATCH request to API while calling the saveChanges() method in context object, which ended up that the API rejected us because some CRM OOB audit fields cannot be modified to null. I used an API tool to send a request that has only the updated field and it worked.

I am using Odata.Client 6.15.0. 

Any thoughts?

Thanks in advance.
